File: atcoder/contest/20181013_gc028/1.rb

n, m = gets.chomp.split(" ").map(&:to_i);

s = gets.strip()
t = gets.strip()

l = n * m /  m.gcd(n)



result = true

for i in 0...n do
    if i * m % n == 0
        k = i * m / n
        if k < t.size() and t[k] != s[i]
            result = false
        end
    end
end

if result
    print(l)
else
    print(-1)
end

File: atcoder/contest/20181013_gc028/1.py
import math

n, m = map(int, input().split())
s = input().strip()
t = input().strip()

l = n * m //  math.gcd(m, n)

result = True

for i in range(0, n-1):
    if i * m % n == 0:
        k = i * m // n
        if k < len(t) and t[k] != s[i]:
            result = False

if result:
    print(l)
else:
    print(-1)
